src/main/java/hudson/plugins/tfs/browsers/TeamSystemWebAccessBrowser.java

```diff
@@ -47,44 +47,49 @@ private String getBaseUrlString(ChangeSet changeSet) throws MalformedURLExceptio
         if (url != null) {
             baseUrl = DescriptorImpl.getBaseUrl(url);
         } else {
-            baseUrl = String.format("%s/", getServerConfiguration(changeSet)); 
+            String scmUrl = getServerConfiguration(changeSet);
+            if (scmUrl.endsWith("/")) {
+              baseUrl = scmUrl;
+            } else {
+              baseUrl = String.format("%s/", scmUrl);
+            }
         }
         return baseUrl;
     }
 
     /*
-     * http://tswaserver:8090/cs.aspx?cs=99
+     * Gets the link to a specific change set.
+     * E.g. http://tswaserver:8090/_versionControl/changeset/99

```

Comment:
Would you mind updating the example URLs (and test cases, if possible) to look like the real-world values (and avoid TSWA-style URLs)?
For example: `https://vsts.example.com/DefaultCollection/` and `http://tfs.example;com:8080/tfs/DefaultCollection/`